packages/web/src/pages/dashboard-page/components/ArtistContentSection.tsx

```diff
@@ -0,0 +1,461 @@
+  key: string
+  name: string
+  date: string
+  time?: number
+  saves: number
+  reposts: number
+  plays: number
+}
+
+export type DataSourceAlbum = Collection & {
+  key: string
+  name: string
+  date: string
+  saves: number
+  reposts: number
+}
+
+enum Pills {
+  TRACKS,
+  ALBUMS
+}
+
+enum TrackFilters {
+  PUBLIC = 'Public',
+  PREMIUM = 'Premium',
+  SPECIAL_ACCESS = 'SpecialAccess',
+  COLLECTIBLE_GATED = 'CollectibleGated',
+  HIDDEN = 'Hidden'
+}
+
+enum AlbumFilters {
+  PUBLIC = 'Public',
+  PREMIUM = 'Premium',
+  HIDDEN = 'Hidden'
+}
+
+const formatAlbum = (album: Collection): DataSourceAlbum => {
+  return {
+    ...album,
+    key: String(album.playlist_id),
+    name: album.playlist_name,
+    date: album.created_at,
+    saves: album.save_count,
+    reposts: album.repost_count
+  }
+}
+
+type ArtistContentSectionProps = {
+  tracks: DataSourceTrack[]
+  account: User
+}
+
+export const ArtistContentSection = ({
+  tracks,
+  account
+}: ArtistContentSectionProps) => {
+  const dispatch = useDispatch()
+  const goToRoute = useGoToRoute()
+  const [filterText, setFilterText] = useState('')
+  const [selectedPill, setSelectedPill] = useState(Pills.TRACKS)
+  const [selectedTrackFilter, setSelectedTrackFilter] =
+    useState<Nullable<TrackFilters>>(null)
+  const [selectedAlbumFilter, setSelectedAlbumFilter] =
+    useState<Nullable<AlbumFilters>>(null)
+  const isTracks = selectedPill === Pills.TRACKS
+  const tracksStatus = useSelector(getDashboardTracksStatus)
+
+  const albumsUnformatted = useSelector(getAccountAlbums) ?? []
+  const albums = albumsUnformatted?.map((album) => formatAlbum(album))
+  const shouldShowPills = tracks.length && albums.length
+
+  const handleFilterChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const val = e.target.value
+    setFilterText(val)
+  }
+
+  const handleSelectFilter = (value: string) => {
+    if (isTracks) {
+      setSelectedTrackFilter(value as TrackFilters)
+    } else {
+      setSelectedAlbumFilter(value as AlbumFilters)
+    }
+  }
+
+  const onClickRow = useCallback(
+    (record: any) => {
+      if (!account) return
+      goToRoute(record.permalink)
+    },
+    [account, goToRoute]
+  )
+
+  const handleFetchPage = useCallback(
+    (page: number) => {
+      dispatch(
+        fetchTracks({ offset: page * tablePageSize, limit: tablePageSize })
+      )
+    },
+    [dispatch]
+  )
+
+  const onClickPill = useCallback(
+    (pill: Pills) => {
+      setSelectedPill(pill)
+      setFilterText('')
+
+      // Reset filter button state when switching content types
+      if (!isTracks && pill === Pills.TRACKS) {
+        setSelectedAlbumFilter(null)
+      } else if (isTracks && pill === Pills.ALBUMS) {
+        setSelectedTrackFilter(null)
+      }
+    },
+    [isTracks]
+  )
+
+  const {
+    hasOnlyOneTrackSection,
+    publicTracks,
+    specialAccessTracks,
+    hiddenTracks,
+    premiumTracks,
+    collectibleGatedTracks
+  } = useMemo(() => {
```

Comment:
nice use of useMemo!

Maybe we could factor this out into a useCallback instead to keep this component's body a bit shorter?